### PR TITLE
Dit 2339

### PR DESCRIPTION
**Pull request by warrenskiwork1979**

### CODE REVIEW

File: app/controllers/CheckYourAnswersController.scala [hunk 1]
```diff
@@ -16,73 +16,144 @@
...
+        ).flatten
       )
     )
 
     Ok(check_your_answers(appConfig, sections))
   }
 
+  def onSubmit(): Action[AnyContent] = (identify andThen getData).async { implicit request: OptionalDataRequest[_] =>
+
+    val answers = request.userAnswers.get // TODO: we should not call `get` on an Option
```
Comment (DavidGregory084) [inline app/controllers/CheckYourAnswersController.scala:107]: If we add `andThen requireData` at the start of the action then we won't have to do the `.get` above

**Review by DavidGregory084 — APPROVED** [review]
Looks good to me 👍

File: app/controllers/CheckYourAnswersController.scala [hunk 2]
```diff
@@ -16,73 +16,144 @@
...
-import service.CountriesService
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import service.{CasesService, CountriesService, FileService}
+import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 import utils.CheckYourAnswersHelper
 import viewmodels.AnswerSection
 import views.html.check_your_answers
 
+import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (DavidGregory084) [inline app/controllers/CheckYourAnswersController.scala:39]: Please can we get rid of this use of the global ExecutionContext and add an `implicit ec` to the controller instead? You can see examples in the other controllers

File: app/controllers/CheckYourAnswersController.scala (hunk 1 above)
Comment (DavidGregory084) [reply app/controllers/CheckYourAnswersController.scala:107]: Then the type of the `implicit request` above will be `DataRequest[_]` instead

Comment (warrenskiwork1979) [reply app/controllers/CheckYourAnswersController.scala]: We've added those changes David

Comment (DavidGregory084) [reply app/controllers/CheckYourAnswersController.scala]: Awesome! 👍 Looks good to go

**Pull request merged**